packages/@ama-sdk/core/src/plugins/session-id/session-id.request.ts

```diff
@@ -34,22 +35,28 @@ export class SessionIdRequest implements RequestPlugin {
    */
   public sessionId: string;
 
+  /**
+   * Logger (optional, fallback to console logger if undefined)
+   * @protected
+   */
+  protected logger?: Logger;
+
   /**
    * Constructor.
    * @param sessionIdHeader The request header in which the ID will be added.
    * @param activateRequestId Indicates if the request ID part should be added to the ID.
+   * @param logger Logger (optional, fallback to console logger if undefined)
    */
-  constructor(sessionIdHeader = 'Ama-Client-Ref', activateRequestId = true) {
+  constructor(sessionIdHeader = 'Ama-Client-Ref', activateRequestId = true, logger?: Logger) {
+    this.logger = logger;
```

Comment:
Maybe add a comment to say that it should be done first as `generateSessionId` use the logger
to avoid anyone moving that latter

Reply:
I also wonder if the `this.sessionId = this.generateSessionId();` should not be moved to the `load` function with `this.sessionId ||= this.generateSessionId();` which make more sense and would avoid providing this *additional loader*